packages/grid/x-data-grid-pro/src/internals/utils/tree/buildRowTree.ts

```diff
@@ -170,13 +170,32 @@ export const buildRowTree = (params: BuildRowTreeParams): GridRowGroupingResult
   for (let i = 0; i < ids.length; i += 1) {
     const rowId = ids[i];
     const tempNode = tempTree[rowId];
-    const node: GridRowTreeNodeConfig = {
+    const nodeWithoutChildrenExpansion: GridRowTreeNodeConfig = {
       ...tempNode,
       children: tempNode.children ? Object.values(tempNode.children) : undefined,
     };
-    const childrenExpanded = isGroupExpandedByDefault(node);
+    const node = {
+      ...nodeWithoutChildrenExpansion,
+      childrenExpanded: isGroupExpandedByDefault(nodeWithoutChildrenExpansion),
+    };
 
-    tree[rowId] = { ...node, childrenExpanded };
+    const previousNode = params.previousTree?.[node.id];
+    const shouldReUsePreviousNode =
+      previousNode &&
+      previousNode.isAutoGenerated === node.isAutoGenerated &&
+      previousNode.parent === node.parent &&
+      previousNode.groupingKey === node.groupingKey &&
+      previousNode.groupingField === node.groupingField &&
+      previousNode.depth === node.depth &&
+      previousNode.childrenExpanded === node.childrenExpanded &&
+      previousNode.children?.length === node.children?.length &&
+      previousNode.children?.every((childId, index) => node.children?.[index] === childId);
```

Comment:
I wonder if checking all children again is not more painful than a new rerender.

Reply:
It is also used in the tree data cell to know the amount of children.
Users can always use the `apiRef` to get the rowNode if we remove it from the prop.
I don't know what would be the best behavior here.